app/controllers/application_controller.rb

```diff
@@ -19,4 +21,18 @@ def admin_authenticated
   def require_user
     render file: '/public/404' unless current_user
   end
+
+  def store_location
+    return unless request.get?
+    if (request.path != "/auth/google_oauth2" &&
+        request.path != "/auth/google_oauth2/callback" &&
+        request.path != "/logout" &&
+        !request.xhr?) # don't store ajax calls
```

Comment:
Consider breaking this out into a private method. Something like `if storable?(request)`. All of the conditions are confusing and the oauth strings seem arbitrary or magical.